### ISSUE TITLE
Pylance not completing suggestions for first party imports

### ISSUE THREAD

**Issue by engnatha** [microsoft/pylance-release#4678]
## Environment data

-   Language Server version: 2023.7.40
-   OS and version: Ubuntu 20.04
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.10

## Repro Steps

Suggestions do appear when typing something like `from <X>.<Y>` as seen below
![image](https://github.com/microsoft/pylance-release/assets/69609716/c05abb2c-c023-4ce3-a2d5-2dacf60fc5d7)
Type out `from <X> import <Y>` in absolute import style and observe module not suggested/recognized until fully typed out.
![image](https://github.com/microsoft/pylance-release/assets/69609716/4756bab1-18fa-4f73-b590-c210d28dc7f0)
But when the module name is fully typed out, it is recognized.
![image](https://github.com/microsoft/pylance-release/assets/69609716/f9a1380c-007f-4379-985e-f7269b962620)
This only happens for first party source code. Third party modules are suggested properly.

## Expected behavior
First party modules should be suggested for autocomplete when using absolute imports in `from <X> import <Y>` style.

## Actual behavior

See above

## Logs

```
2023-08-01 23:20:09.614 [info] [Info  - 11:20:09 PM] (1363060) Pylance language server 2023.7.40 (pyright b917b7a5) starting
2023-08-01 23:20:09.614 [info] [Info  - 11:20:09 PM] (1363060) Server root directory: /home/nathanael/.vscode/extensions/ms-python.vscode-pylance-2023.7.40/dist
2023-08-01 23:20:09.623 [info] [Info  - 11:20:09 PM] (1363060) Starting service instance "monorepo"
2023-08-01 23:20:09.656 [info] [Info  - 11:20:09 PM] (1363060) Setting pythonPath for service "monorepo": "/home/nathanael/devel/monorepo/dist/export/python/virtualenvs/python-default/3.8.10/bin/python"
2023-08-01 23:20:09.657 [info] [Info  - 11:20:09 PM] (1363060) Loading configuration file at /home/nathanael/devel/monorepo/pyrightconfig.json
2023-08-01 23:20:09.709 [info] [Info  - 11:20:09 PM] (1363060) Assuming Python version 3.8
2023-08-01 23:20:09.709 [info] [Warn  - 11:20:09 PM] (1363060) The venvPath has been specified in both the config file and the client settings. The value in the config file (/home/nathanael/devel/monorepo) will take precedence
2023-08-01 23:20:09.709 [info] [Warn  - 11:20:09 PM] (1363060) The useLibraryCodeForTypes has been specified in both the config file and the client settings. The value in the config file (true) will take precedence
2023-08-01 23:20:10.329 [info] [Info  - 11:20:10 PM] (1363060) Found 3650 source files
2023-08-01 23:20:10.815 [info] [Info  - 11:20:10 PM] (1363060) Background analysis(1) root directory: /home/nathanael/.vscode/extensions/ms-python.vscode-pylance-2023.7.40/dist
2023-08-01 23:20:10.816 [info] [Info  - 11:20:10 PM] (1363060) Background analysis(1) started
2023-08-01 23:20:11.172 [info] [Info  - 11:20:11 PM] (1363060) Indexer background runner(2) root directory: /home/nathanael/.vscode/extensions/ms-python.vscode-pylance-2023.7.40/dist (index)
2023-08-01 23:20:11.172 [info] [Info  - 11:20:11 PM] (1363060) Indexing(2) started
2023-08-01 23:20:12.433 [info] [Info  - 11:20:12 PM] (1363060) scanned(2) 472 files over 1 exec env
2023-08-01 23:20:17.999 [info] [Info  - 11:20:17 PM] (1363060) [IDX(2)] Long operation: index execution environment /home/nathanael/devel/monorepo (5397ms)
2023-08-01 23:20:18.000 [info] [Info  - 11:20:18 PM] (1363060) [IDX(2)] Long operation: index packages /home/nathanael/devel/monorepo (5430ms)
2023-08-01 23:20:18.000 [info] [Info  - 11:20:18 PM] (1363060) indexed(2) 342 files over 1 exec env
2023-08-01 23:20:18.046 [info] [Info  - 11:20:18 PM] (1363060) Indexing finished(2).
```

**Comment by heejaechang** [comment]
it has been broken for long time.

**Comment by heejaechang** [comment]
actually, I think we should show all modules we can use even for `namespace package`

**Comment by engnatha** [comment]
We only do namespace packages. `__init__.py` files are placed sparsely around because tools like `sphinx` sometimes need them to traverse namespaces correctly when rendering.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.9.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023911-6-September-2023-PreRelease)
